File: 2015/Day 1/Day1.py
# Part A
# answer = 0
# with open("input.txt") as f:
#     for char in f.readline():
#         if char == '(':
#             answer += 1
#         else:
#             answer -= 1
# print(answer)

# Part B
answer = 0
count = 0
with open("input.txt") as f:
    for char in f.readline():
        count += 1
        if char == '(':
            answer += 1
        else:
            answer -= 1
        if answer == -1:
            print(count)
            break

File: 2021/Day 1/Day1.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
	FILE *file;
	int count = 0;
	int first, second, third, sum1, sum2;

	file = fopen("input.txt", "r");
	if (file == NULL)
	{
		printf("File not found.");
		return 0;
	}

	//Part A

	// fscanf (file, "%d", &first);
	// while (!feof(file))
	// {
	// 	fscanf(file, "%d", &second);
	// 	if (second > first)
	// 	{
	// 		count++;
	// 	}
	// 	first = second;
	// }
	// fclose(file);
	// printf("%d", count);

	//Part B

	fscanf(file, "%d", &first);
	fscanf(file, "%d", &second);
	fscanf(file, "%d", &third);
	sum1 = first + second + third;

	while (!feof(file))
	{
		first = second;
		second = third;
		fscanf(file, "%d", &third);
		sum2 = first + second + third;
		if (sum1 < sum2)
		{
			count++;
		}
		sum1 = sum2;
	}
	fclose(file);
	printf("%d", count);
}